pkg/storage/errors.go

```diff
@@ -96,15 +114,24 @@ func IsUnreachable(err error) bool {
 
 // IsTestFailed returns true if and only if err is a write conflict.
 func IsTestFailed(err error) bool {
-	return isErrCode(err, ErrCodeResourceVersionConflicts)
+	return isErrCode(err, ErrCodeResourceVersionConflicts, ErrCodeInvalidUID)
```

Comment:
@timothysc maybe I misunderstand your question, UID is a field in the ObjectMeta, so it's in almost in every API object. https://github.com/kubernetes/kubernetes/blob/master/pkg/api/v1/types.go#L116